### DISCUSSION TITLE
Two new instances don't appears from GUI

### DISCUSSION THREAD

**Discussion by acappelletti** [trimble-oss/dba-dash#791] [General]
Hi,
I added 2 new connections but I don't see the instances appearing in the GUI tree. I'm not using WMI as I'm behind VPN and had errors in the logs... so I remoed WMI.

I checked also 
![image](https://github.com/trimble-oss/dba-dash/assets/43847440/141b635e-549f-4051-a677-a22ebe185752)

but are errors of restart of instance...
Also I checked che collector log... and all seems fine.

2023-12-05 09:52:00.672 +01:00 [INF] Collect ObjectExecutionStats, CPU, IOStats, Waits, AzureDBResourceStats, AzureDBElasticPoolResourceStats, SlowQueries, PerformanceCounters, JobHistory, RunningQueries, DatabasesHADR, AvailabilityGroups, AvailabilityReplicas, MemoryUsage, RunningJobs from instance XXXXXXX completed in 621.5 ms <21>
2023-12-05 09:52:00.853 +01:00 [INF] Write to destination HP-PROBOOK|DBADashDB from XXXXXXX completed in 181.4 ms <21>

but I don't see the 2 new instances into table
[DBADashDB].[dbo].[Instances]

ALEN

**Comment by acappelletti** [comment]
I noticed a strange behavior... the table I exposed already contained the HOSTNAME of the two servers that I had now added with the IPs and they were disabled. I forced the reactivation with a
```
  UPDATE [DBADashDB].[dbo].[Instances]
   SET IsActive = 1
   WHERE [InstanceID] IN (3,4)
```
and they reappeared in the GUI with the updated values.

I expected that having re-entered them into the connection string with the IPs they would be treated as new servers.
Furthermore, I added the one already present in the table to the "connection ID" field for greater security (literal service name).

Ultimately my question is: but if I delete a connection string (before it was literal) and then reinsert it with IP, is it correct that the instance is not added to the GUI? I had to intervene manually... I'm sure the data is correct since I checked the snapshots

THNAKs ALEN

**Comment by DavidWiseman** [comment]
The ConnectionID is what uniquely identifies a connection and maps the collected data to a row in the dbo.Instances table.  If you don't explicitly define this it will be set to the value of `@@SERVERNAME` for normal SQL connections.  If you change the connection string to use an IP address or a different alias - the ConnectionID wouldn't change so it will still appear as the same instance in the repository DB.  If the instance has been marked deleted that would explain why it didn't show up.  You can always restore an instance in the GUI using the "Manage Instances" dialog and clicking the "Restore" link.  
If you want to have more control over how an instance is identified, set the ConnectionID in the configuration tool.  This should be unique for each connection.  

Note: A restart of the service is required to pick up any config changes.

Hope this helps.

**Reply by acappelletti** [reply]
Yes I followed the GUI reactivation after... now it's OK.
THANKS David!
